packages/core/lib/src/data/build_bits.dart

```diff
@@ -204,7 +204,7 @@ abstract class BuildTree extends BuildBit<Null, Iterable<Widget>> {
   BuildBit addWhitespace() => add(_TextWhitespace(this));
 
   /// Adds a string of text.
-  TextBit addText(String data) => add(TextBit(this, data));
+  TextBit addText(String data) => add(TextBit(this, data)) as TextBit;
```

Comment:
Can we change the `add` signature into something like:

```dart
T add<T extends BuildBit>(T bit)
```

So the casting is no longer required here?